### DISCUSSION TITLE
Add a more meaningful oauth failure message and the ability to add a custom message.

### DISCUSSION THREAD

**Discussion by HumairAK** [grafana/grafana#42162] [Platform]
We currently use Grafana Oauth [role_atribute_path](https://grafana.com/docs/grafana/latest/auth/generic-oauth/#role-mapping), we do something like: 

```yaml
      role_attribute_path: >-
        contains(groups[*], 'cluster-admins') && 'Admin' ||
        contains(groups[*], 'neu-students')   && 'Editor' ||
        'Deny'
```
When a user is not in the required group 'Deny' is returned, which is rejected as it is not a valid role. On the website they are then greeted with an error like the one here: 

![image](https://user-images.githubusercontent.com/10904967/143075983-fa4c2ce0-5719-45ca-a29e-46b0d2efcf2d.png)

In Plaintext:
```
Grafana / Server Error
Sadly something went wrong
login.OAuthLogin(get info from generic_oauth)

Check the Grafana server logs for the detailed error message.
```

It would be nice if the message here was a bit more helpful. `Sadly something went wrong` is not helpful, neither is `login.OAuthLogin(get info from generic_oauth)` or `login.OAuthLogin(get info from generic_oauth)` as users do not have access to the configuration/log files for grafana. So it is not clear to them that they are failing to login because they don't have the required role. 

It would be better if the message was a bit more clear like "Authentication attempt fail via Oauth", for instance. And it would be even better if we can specify our own custom message (with links) for the user so we may instruct them on how to resolve this issue, or where to go next.

**Comment by wodzuu** [comment]
This problem still occurs in version 9.1.2. Returning `null` value from JMESPath expression in `GF_AUTH_GENERIC_OAUTH_ROLE_ATTRIBUTE_PATH` results in confusing error page as shown by @HumairAK .
In the logs we can see `level=error msg="login.OAuthLogin(get info from generic_oauth)" error="invalid role"` so it is handled as a specific error, not a generic one. Could we at least make the error message say `login.OAuthLogin(invalid role)` instead of `login.OAuthLogin(get info from generic_oauth)` ?

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
